framework/DataObjects/XHistorySet.py

```diff
@@ -265,10 +265,21 @@ def _toCSV(self,fileName,start=0,**kwargs):
     self._usePandasWriteCSV(fileName,inpData,ordered,keepSampleTag = self.sampleTag in keep,mode=mode)
     ## obtain slices to write subset CSVs
     ordered = list(o for o in self.getVars('output') if o in keep)
+
     if len(ordered):
-      for i in range(len(data[self.sampleTag].values)):
-        filename = subFiles[i][:-4]
-        rlz = data.isel(**{self.sampleTag:i})[ordered].dropna(self.indexes[0])
-        self._usePandasWriteCSV(filename,rlz,ordered,keepIndex=True)
+      # hierarchical
+      if not self.hierarchical and 'RAVEN_isEnding' in self.getVars():
+        for i in range(len(fullData)):
+          mode = 'w'
```

Comment:
Some comments here explaining why we change to write/append mode, and the thinking in general, would be helpful for next time through.

Reply:
added